src/libraries/System.Private.CoreLib/src/System/Text/EncodingTable.cs

```diff
@@ -105,18 +106,23 @@ private static int InternalGetCodePageFromName(string name)
         // Return a list of all EncodingInfo objects describing all of our encodings
         internal static EncodingInfo[] GetEncodings()
         {
-            EncodingInfo[] arrayEncodingInfo = new EncodingInfo[s_mappedCodePages.Length];
+            List<EncodingInfo> encodingInfos = new List<EncodingInfo>(s_mappedCodePages.Length);
```

Comment:
>encodingInfos [](start = 31, length = 13)

can we just use array as before and decide about the length according to the config switch value? this will help avoid allocating the List.